File: java/src/main/java/codingdojo/BasicBuff.java
package codingdojo;

public class BasicBuff implements Buff {

    private final float soak;
    private final float damage;

    public BasicBuff(float soak, float damage) {
        this.soak = soak;
        this.damage = damage;
    }

    @Override
    public float soakModifier() {
        return soak;
    }

    @Override
    public float damageModifier() {
        return damage;
    }
}

File: typescript/src/BasicBuff.ts
import {Buff} from './Buff';

export class BasicBuff implements Buff {

    public constructor(private _soak: number,
                       private _damage: number){
    }

    public get soakModifier(): number{
        return this._soak;
    }

    public get damageModifier(): number{
        return this._damage;
    }
}
